test/python/algorithms/test_numpy_minimum_eigen_solver.py

```diff
@@ -119,6 +130,31 @@ def test_cme_1q(self, op):
         result = algo.compute_minimum_eigenvalue(operator=op)
         self.assertAlmostEqual(result.eigenvalue, -1)
 
+    def test_cme_aux_ops_dict(self):
+        """Test dictionary compatibility of aux_operators"""
+        # Start with an empty dictionary
+        algo = NumPyMinimumEigensolver()
+        result = algo.compute_minimum_eigenvalue(operator=self.qubit_op, aux_operators={})
+        self.assertAlmostEqual(result.eigenvalue, -1.85727503 + 0j)
+        self.assertIsNone(result.aux_operator_eigenvalues)
+
+        # Add aux_operators dictionary and go again
+        result = algo.compute_minimum_eigenvalue(
+            operator=self.qubit_op, aux_operators=self.aux_ops_dict
+        )
+        self.assertAlmostEqual(result.eigenvalue, -1.85727503 + 0j)
+        self.assertEqual(len(result.aux_operator_eigenvalues), 2)
+        np.testing.assert_array_almost_equal(result.aux_operator_eigenvalues["aux_op1"], [2, 0])
+        np.testing.assert_array_almost_equal(result.aux_operator_eigenvalues["aux_op2"], [0, 0])
+
+        # Add None and zero operators and go again
+        extra_ops = {"None_op": None, "zero_op": 0, **self.aux_ops_dict}
+        result = algo.compute_minimum_eigenvalue(operator=self.qubit_op, aux_operators=extra_ops)
+        self.assertAlmostEqual(result.eigenvalue, -1.85727503 + 0j)
+        self.assertEqual(len(result.aux_operator_eigenvalues), 2)
+        np.testing.assert_array_almost_equal(result.aux_operator_eigenvalues["aux_op1"], [2, 0])
+        np.testing.assert_array_almost_equal(result.aux_operator_eigenvalues["aux_op2"], [0, 0])
+
 
```

Comment:
@mrossinek I've only added a test for the NumPyMinimumEigenSolver since I did not feel comfortable with adding one for the VQE due to the probabilistic nature. Does this suffice?

Reply:
As a comment we have the algorithm_globals random seed, plus there are simulator and transpiler seeds. You can see these set on a number of tests where random functions are involved to allow the tests to be reproducible. And for VQE the statevector simulation can be used too to get the ideal outcome.

Reply:
I think we also need some tests for VQE because it does not reuse the entire code from the NumPyMinimumEigensolver. Steve mentioned above how you can avoid issues with the random nature of the tests.